src/psygnal/_group.py

```diff
@@ -251,53 +256,74 @@ class MySignals(SignalGroup):
     """
 
     _psygnal_signals: ClassVar[Mapping[str, Signal]]
-    _psygnal_instances: dict[str, SignalInstance]
     _psygnal_uniform: ClassVar[bool] = False
+    _psygnal_name_conflicts: ClassVar[set[str]]
+
+    _psygnal_instances: dict[str, SignalInstance]
 
     def __init__(self, instance: Any = None) -> None:
         cls = type(self)
-        if not hasattr(cls, "_psygnal_signals"):  # pragma: no cover
+        if not hasattr(cls, "_psygnal_signals"):
             raise TypeError(
                 "Cannot instantiate `SignalGroup` directly.  Use a subclass instead."
             )
 
         self._psygnal_instances = {
-            name: sig.__get__(self, cls) for name, sig in cls._psygnal_signals.items()
+            name: (
+                sig._create_signal_instance(self)
```

Comment:
:eyes: @getzze 

here was my solution for using the standard Signal.__get__ descriptor protocol for non-conflicting names, and a (non-Signal-cached) local instance for conflicting names.